CleanProFinder/CleanProFinder.Server/Services/Implementations/ValidationService.cs

```diff
@@ -0,0 +1,29 @@
+﻿using CleanProFinder.Server.Services.Interfaces;
+using CleanProFinder.Shared.Errors.Base;
+using CleanProFinder.Shared.ServiceResponseHandling;
+using FluentValidation;
+using System.Reflection;
+
+
+namespace CleanProFinder.Server.Services.Implementations
+{
+    public class ValidationService : IValidationService
+    {
+        public async Task<ServiceResponse> ValidateAsync<T>(T item)
+        {
+            var validatorType = Assembly.GetExecutingAssembly().GetTypes()
+                .FirstOrDefault(t => t.BaseType != null && t.BaseType.IsGenericType &&
+                                     t.BaseType.GetGenericTypeDefinition() == typeof(AbstractValidator<>) &&
+                                     t.BaseType.GetGenericArguments().Any(genericArg => genericArg == typeof(T)));
+
```

Comment:
why can not we just use 
var validatorType = Assembly.GetExecutingAssembly().GetTypes().FirstOrDefault(t => typeof(AbstractValidator<T>). IsAssignableFrom(T)
?

Reply:
fixed